pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ProducerHandler.java

```diff
@@ -315,30 +351,117 @@ protected ProducerBuilder<byte[]> getProducerBuilder(PulsarClient client) {
             builder.sendTimeout(Integer.parseInt(queryParams.get("sendTimeoutMillis")), TimeUnit.MILLISECONDS);
         }
 
-        if (queryParams.containsKey("batchingEnabled")) {
-            builder.enableBatching(Boolean.parseBoolean(queryParams.get("batchingEnabled")));
+        if (queryParams.containsKey("messageRoutingMode")) {
+            checkArgument(
+                    Enums.getIfPresent(MessageRoutingMode.class, queryParams.get("messageRoutingMode")).isPresent(),
+                    "Invalid messageRoutingMode %s", queryParams.get("messageRoutingMode"));
+            MessageRoutingMode routingMode = MessageRoutingMode.valueOf(queryParams.get("messageRoutingMode"));
+            if (!MessageRoutingMode.CustomPartition.equals(routingMode)) {
+                builder.messageRoutingMode(routingMode);
+            }
```

Comment:
Is it a part of the proposal?

Reply:
Ah, it is the [original design](https://github.com/apache/pulsar/pull/862), I've not changed it.

See: https://github.com/apache/pulsar/pull/862

Reply:
Got it.